src/ray/core_worker/store_provider/plasma_store_provider.cc

```diff
@@ -287,8 +294,8 @@ Status CoreWorkerPlasmaStoreProvider::Get(
   int64_t total_size = static_cast<int64_t>(object_ids.size());
   for (int64_t start = 0; start < total_size; start += batch_size) {
     batch_ids.clear();
-    for (int64_t i = start; i < batch_size && i < total_size; i++) {
-      batch_ids.push_back(id_vector[start + i]);
+    for (int64_t i = start; i < start + batch_size && i < total_size; i++) {
+      batch_ids.push_back(id_vector[i]);
```

Comment:
This is the main fix and other changes are only for making things testable. If batch_size is n, then the outer loop iterates with `start` values: 0, n, 2n, etc. The inner loop should process indices from `start` to `start + batch_size - 1` (or until `total_size`). But the current (master) condition `i < batch_size` means it only processes indices 0 to batch_size-1, regardless of the start value. Also, `id_vector[start + i]` should be `id_vector[i]` since `i` already starts from `start`.